### DISCUSSION TITLE
Question about WRITE_HOLD_REGISTER

### DISCUSSION THREAD

**Discussion by Thierrybelg** [eModbus/eModbus#290] [General]
Hello,
Testing this great library since some days.
Goal is to control Alfen charging station.
Read register is ok : err = MB.addRequest((uint32_t)lastMillis, 2, READ_HOLD_REGISTER, 1210, 2);
When I try to write this register, I have an error : err = MB.addRequest((uint32_t)lastMillis, 1, WRITE_HOLD_REGISTER, 1210, 10.00);
--> Error response: 02 - Illegal data address token: 10002
The data is a Float32 on register 1210-1211

Any help appreciate :-)
Sorry for my bad english, i'm french.

Thierry

**Comment by Miq1** [comment]
Hi, WRITE_HOLD_REGISTER (function code 0x06) is only able to write a single register of 16 bits - so tells us the Modbus standard. Your float value is 32 bits, hence the error.

You will want to use WRITE_MULT_REGISTERS (function code 0x10) instead, that is able to write up to 126 registers in a row.

**Comment by Thierrybelg** [comment]
Thanks for your help :-)
Where can I find document for WRITE_MULT_REGISTERS ?

**Reply by Miq1** [reply]
See the official Modbus specs at [the Modbus.org site](https://www.modbus.org/specs.php).

The eModbus call is like 
```
addRequest(uint32_t token, uint8_t serverID, WRITE_MULT_REGISTERS, uint16_t address, uint16_t registerCount, uint8_t byteCount, uint16_t data[]);
```

There is the ``RTU16example`` that is demonstrating it.
